### PR TITLE
feat: endpoint for creating/updating exams

### PR DESCRIPTION
**Pull request by m-pitera**
**Description:**
Adds a new endpoint to create and update exams. This endpoint should take a list of all course exams and match the behavior as if calling [register_proctored_exams](https://github.com/openedx/edx-platform/blob/08eb1551b2d12fce876f0d6e7e6fdb2f83877b96/cms/djangoapps/contentstore/proctoring.py#L28) in edx-platform.

**JIRA:**
[MST-1521](https://2u-internal.atlassian.net/browse/MST-1521)

### CODE REVIEW

**Comment by Zacharis278** [comment]
Opted to move this to its own view since there's some subtle differences in the object we are serializing here.

File: edx_proctoring/views.py [hunk 1]
```diff
@@ -478,6 +479,86 @@ def get(self, request, exam_id=None, course_id=None, content_id=None):  # pylint
...
+                        is_proctored=request_exam.get('is_proctored', None),
+                        is_practice_exam=request_exam.get('is_practice_exam', None),
+                        external_id=request_exam.get('external_id', None),
+                        is_active=request_exam.get('is_active', None),
+                        hide_after_due=request_exam.get('hide_after_due', None),
+                        backend=request_exam.get('backend', None),
+                    )
+
+                    write_result.append({'exam_id': exam.id})
+                    msg = f'Updated exam {exam.id}'
```
Comment (ashultz0) [inline edx_proctoring/views.py:518]: do we want exam ID or the content_id which is the lookup key? Or both?

File: edx_proctoring/views.py [hunk 2]
```diff
@@ -478,6 +479,86 @@ def get(self, request, exam_id=None, course_id=None, content_id=None):  # pylint
...
+                        is_proctored=request_exam.get('is_proctored'),
+                        is_practice_exam=request_exam.get('is_practice_exam'),
+                        external_id=request_exam.get('external_id'),
+                        is_active=request_exam.get('is_active'),
+                        hide_after_due=request_exam.get('hide_after_due'),
+                        backend=request_exam.get('backend'),
+                    )
+
+                    write_result.append({'exam_id': exam_id})
+                    msg = f'Created new exam {exam_id}'
```
Comment (ashultz0) [inline edx_proctoring/views.py:536]: same logging question

File: edx_proctoring/views.py [hunk 3]
```diff
@@ -478,6 +479,86 @@ def get(self, request, exam_id=None, course_id=None, content_id=None):  # pylint
...
+            # The remaining exams exist but are not included in the registration payload.
+            # That means this exam should be disabled.
+            # We need to mark these exams as inactive (we don't delete!)
+            for exam in course_exams.values():
+                if exam.is_active:
+                    msg = f'Disabling exam {exam.id}'
+                    LOG.info(msg)
+                    update_exam(
+                        exam.id,
+                        is_proctored=False,
```
Comment (ashultz0) [inline edx_proctoring/views.py:548]: do we need to set is_proctored to false? It was a proctored exam, and now it is inactive but it still was a proctored exam when it was alive.

**Review by ashultz0 — APPROVED** [review]
looks pretty good, a few logging thoughts you can take or leave

I think realizing that registration is not really the same as updating freed you up to make some good twists to the code

File: edx_proctoring/views.py (hunk 1 above)
Comment (Zacharis278) [reply edx_proctoring/views.py:518]: personally, if I'm debugging anything I'm just going to look this up by the database id which is a whole lot cleaner than messing with a big string (content_id)

File: edx_proctoring/views.py (hunk 3 above)
Comment (Zacharis278) [reply edx_proctoring/views.py:548]: actually I don't even know why this is necessary, studio does this today so I just kept the behavior. https://github.com/openedx/edx-platform/blob/6c47a6d0aa0ea490cb84650f624c085565014050/cms/djangoapps/contentstore/proctoring.py#L147

**Pull request merged**